baybe/utils/dataframe.py

```diff
@@ -643,7 +643,24 @@ def filter_df(
            num cat
         2    1   a
         3    1   b
+
+        >>> filter_df(df, pd.DataFrame(), complement=True)
+           num cat
+        0    0   a
+        1    0   b
+        2    1   a
+        3    1   b
+
+        >>> filter_df(df, pd.DataFrame(), complement=False)
+        Empty DataFrame
+        Columns: [num, cat]
+        Index: []
+
     """
+    # Handle special case of empty filter
+    if filter.empty:
```

Comment:
I understand your argument 👍🏼 And potentially there is a more intuitive implementation. But note that this is not about the `empty` part but rather how the function is designed as a whole. I'm open to discussion here and we can still easily adjust since not yet released.

Right now, the "filter" does not define what is `thrown` but what is `kept` a la join. Depending on the expectation, this either *is* or *is not* surprising. In any case, the potential misunderstanding it can bring is that for the join there is no obvious definition or "smaller filter" since reduction in rows has the different effect than reduction in columns.

That is:
* Going from `pd.DataFrame({"a": [0], "b": [0]})` to just `pd.DataFrame({"a": [0]})` results in  **more** output because more matches can be established, in agreement with your logic.
* However, going from `pd.DataFrame({"a": [0, 1]})` to `pd.DataFrame({"a": [0]})` give **less** output, because fewer matches are established.

This is all fine as long as we are clear that filtering is done via join. But it contradicts your logic in the edge case because – I think in the venn sense – a join with an empty dataframe should logically give an empty result. But is in fact undefined in pandas, hence the guard clause.

Happy to hear your thoughts

Reply:
I see, thanks for clarifying. Indeed my suggestions is not relevant then as we do not have the same limes depending on which variable goes to zero (rows or columns), so forget about it

as for naming: I much prefer names like `to_keep` or `to_drop`, it makes it crystal clear whats the role of the variable and imo is better than `filter`

Reply:
I have nothing against renaming to make the role clearer. But how to handle the `complement` argument in this case? I.e. can you give me a suggestions for a good combo for both arguments? Or is `to_keep` combined with `complement` good enough? Not sure if confusing ... 🤔

Reply:
here even more alternatives
- `df_to_keep` or `df_to_drop`
- `invert`, `invert_mask` or `invert_merge`

Reply:
Alright, have changed the names, now using `to_keep` and `complement` (I think the latter still works and then it is at least consistent with the `toggle` signature in campaign). Have also updated/refined the docstrings accordingly